src/DateTime.cc

```diff
@@ -474,7 +477,23 @@ struct tm DateTime::gmtime(Timescale scale) const {
         nsecs -= frac;
     }
     time_t secs = static_cast<time_t>(nsecs / LL_NSEC_PER_SEC);
+
+    bool inLeap = false;
+    if (scale == UTC) {
+        for (size_t i = 0; i < leapSecTable.size(); ++i) {
+            if (_nsecs >= leapSecTable[i].whenTai - 1000000000 &&
+                _nsecs < leapSecTable[i].whenTai) {
+                inLeap = true;
+                break;
+            }
+        }
+        if (inLeap) --secs;
+    }
```

Comment:
_nsec is nanoseconds in UTC. Why are you comparing that against TAI entries in the leap second table?